src/SignalRServiceExtension/Bindings/SignalRIAsyncCollectorExtensions.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.SignalR.Management;
+using Microsoft.Azure.WebJobs.Extensions.SignalRService;
+
+namespace Microsoft.Azure.WebJobs
+{
+    public static class SignalRIAsyncCollectorExtensions
+    {
+        public static ValueTask<IServiceHubContext> GetServiceHubContextAsync(this IAsyncCollector<SignalRMessage> collector, string hubName) =>
```

Comment:
Add one for `IAsyncCollector<SignalRGroupAction>` as well?

Reply:
It's cool that this works. I definitely prefer a fully-functional IServiceHubContext over my PR.

I do think it's weird to make this an extension method on `IAsyncCollector<SignalRMessage>` considering GetServiceHubContextAsync doesn't even use the collector.

I would probably just add a public static `ServiceHubContextStore.GetServiceHubContextAsync` method that does the same thing and calls StaticServiceHubContextStore.ServiceHubContextStore.GetOrAddAsync(hubName).

Reply:
I think a generic type `IAsyncCollector<T>.GetServiceHubContextAsync(...)` is better, since extension methods show up in intellisense is good like you said.

Reply:
It would be nice to be able to use the IServiceHubContext even if you didn't take an `IAsyncCollector<T>` parameter. I could see an argument for having both an extension method and normal static method.

Reply:
Sounds good. C# users don't need to use the binding `SignalRAttribute` if they like.